### PR TITLE
V1.6 user management

### PR DESCRIPTION
**Pull request by josheide**
This version brings in user / password management but without fundamentally changing things on the task management side of things

### CODE REVIEW

File: userList.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Tomi - asd
```
Comment (balazskiss) [inline userList.txt:1]: This should be in gitignore file

File: src/Main.java [hunk 2]
```diff
@@ -1,207 +1,425 @@
...
+                                break;
+                            } else {
+                                System.out.println("Sorry, I did not catch that!");
                             }
                         }
+                    } else {
+                        System.out.println("Invalid password. Please try again.");
                     }
+                } else{
+            System.out.println("Sorry, I could not find " + userInputName );
```
Comment (balazskiss) [inline src/Main.java:330]: code indentation is off here

File: src/Main.java [hunk 3]
```diff
@@ -1,207 +1,425 @@
...
 
-        String filePathWelcome = "welcomeText.txt";
-        try (BufferedReader reader = new BufferedReader(new FileReader(filePathWelcome))) {
-            String line;
-            while ((line = reader.readLine()) != null) {
-                System.out.println(line);
             }
-        } catch (IOException e) {
-            System.out.println("An error occurred while reading the file: " + e.getMessage());
+//            else if (userInputName.isEmpty()) {
```
Comment (balazskiss) [inline src/Main.java:46]: remove commented code

**Review by balazskiss — CHANGES_REQUESTED** [review]

File: src/Main.java [hunk 4]
```diff
@@ -1,207 +1,425 @@
...
+                    } else {
+                        System.out.println("Invalid password. Please try again.");
                     }
+                } else{
+            System.out.println("Sorry, I could not find " + userInputName );
                 }
-            } else if (userInput.equals("exit")) {
+            }
+        }
+    public static void deleteUser() {
```
[the remark this answers is not in the document]
Comment (josheide) [reply src/Main.java:303]: Done. This was a bit trickier than I thought as I ran into a ConcurrentModificationException.

File: src/Main.java [hunk 5]
```diff
@@ -1,207 +1,425 @@
...
+            if (!userInputName.isEmpty()) {
+                // List tasks for a specific user
+                for (Task element : taskList) {
+                    if (element.assignedToUser.equals(userInputName)) {
+                        System.out.println("- " + element.getId() + " - " + element.name + " - " + (element.isComplete ? "completed" : "incomplete") + " - assigned to " + element.assignedToUser);
+                    }
+                }
+            }
+
+//            else {
```
[the remark this answers is not in the document]
Comment (josheide) [reply src/Main.java]: Done

File: src/Main.java [hunk 6]
```diff
@@ -1,207 +1,425 @@
...
+    static ArrayList<User> userArrayList = new ArrayList<User>();
+    static int nextID = 1;
+    static Scanner scanner = new Scanner(System.in);
 
-        try {
-            if (!todolist.exists()) {
-                todolist.createNewFile();
-            }
-            BufferedReader reader = new BufferedReader(new FileReader(todolist));
+    public static void welcomeMessage(){
```
[the remark this answers is not in the document]
Comment (josheide) [reply src/Main.java]: Done!

File: src/Main.java (hunk 3 above)
Comment (josheide) [reply src/Main.java]: Done!

File: src/Main.java (hunk 2 above)
Comment (josheide) [reply src/Main.java]: Fixed it!

**Review by balazskiss — APPROVED** [review]
nice

**Pull request merged**